CombinatorialGames/Game/Birthday.lean

```diff
@@ -247,7 +261,120 @@ instance small_setOf_birthday_lt (o : NatOrdinal.{u}) : Small.{u} {x // birthday
   apply @small_subset _ _ _ _ (small_setOf_birthday_le o)
   exact fun x (hx : x.birthday < _) ↦ le_of_lt hx
 
--- TODO: short game iff finite birthday
+/-! #### Short games -/
+
+/-- The finset of all games with birthday ≤ n. -/
+noncomputable def birthdayFinset : ℕ → Finset IGame.{u}
+  | 0 => {0}
+  | n + 1 => ((birthdayFinset n).powerset ×ˢ (birthdayFinset n).powerset).map
+    ⟨fun ⟨a, b⟩ => {a | b}ᴵ, fun a b hab => by aesop⟩
+
+theorem mem_birthdayFinset_succ {x : IGame} {n : ℕ} : x ∈ birthdayFinset (n + 1) ↔
+    ∃ l r, (l ⊆ birthdayFinset n ∧ r ⊆ birthdayFinset n) ∧ {l | r}ᴵ = x := by
+  simp [birthdayFinset]
+
+@[simp] theorem birthdayFinset_zero : birthdayFinset 0 = {0} := rfl
+
+
+theorem birthdayFinset_one :
+    birthdayFinset 1 = ⟨[0, 1, -1, ⋆], by aesop (add simp [IGame.ext_iff])⟩ := by
+  ext
+  rw [mem_birthdayFinset_succ]
+  aesop (add simp [IGame.ext_iff])
+
+@[simp]
+theorem card_birthdayFinset (n : ℕ) :
+    (birthdayFinset.{u} (n + 1)).card = 4 ^ (birthdayFinset.{u} n).card := by
+  rw [birthdayFinset, Finset.card_map, Finset.card_product, Finset.card_powerset, ← mul_pow]
+  rfl
+
+theorem mem_birthdayFinset_of_isOption {x y : IGame} {n : ℕ} (hnx : x ∈ birthdayFinset (n + 1))
+    (hy : IsOption y x) : y ∈ birthdayFinset n := by
+  rw [mem_birthdayFinset_succ] at hnx
+  aesop
+
+@[simp]
+theorem mem_birthdayFinset {x : IGame} {n : ℕ} : x ∈ birthdayFinset n ↔ x.birthday ≤ n := by
+  induction n generalizing x with
+  | zero => simp
+  | succ n IH =>
+    simp_rw [mem_birthdayFinset_succ, birthday_le_iff, Finset.subset_iff, Nat.cast_add_one,
+      ← succ_eq_add_one, lt_succ_iff, IH]
+    constructor
+    · aesop
+    · rintro ⟨hl, hr⟩
+      have hxl : x.leftMoves ⊆ birthdayFinset n := by intro y; simp_all
+      have hxr : x.rightMoves ⊆ birthdayFinset n := by intro y; simp_all
+      classical
+      have := Set.fintypeSubset _ hxl
+      have := Set.fintypeSubset _ hxr
+      use x.leftMoves.toFinset, x.rightMoves.toFinset
+      aesop
+
+theorem strictMono_birthdayFinset : StrictMono birthdayFinset := by
+  refine strictMono_nat_of_lt_succ fun n ↦ ⟨fun y hy ↦ ?_, fun h ↦ ?_⟩
+  · rw [mem_birthdayFinset] at *
+    apply hy.trans
+    simp
+  · have := Finset.card_le_card h
+    rw [card_birthdayFinset] at this
+    exact (Nat.lt_pow_self (Nat.one_lt_succ_succ 2)).not_le this
+
+theorem finite_setOf_subposition_of_birthday_lt_omega0 {x : IGame}
```

Comment:
I feel a bit silly only now realizing this, but I don't think we should have these theorems. After all, our goal is to show the `x.birthday < ω` condition is equivalent to `Short x`, and we already have these theorems for `Short`!

I propose `finite_setOf_subposition_of_birthday_lt_omega0` either gets privated or inlined into `short_iff_finite_birthday`. The other three theorems below this can simply be deleted.